zinc/src/main/scala/sbt/internal/inc/LookupImpl.scala

```diff
@@ -39,11 +40,8 @@ class LookupImpl(compileConfiguration: CompileConfiguration, previousSetup: Opti
 
   private val entry = MixedAnalyzingCompiler.classPathLookup(compileConfiguration)
 
-  override def lookupAnalysis(binaryClassName: String): Option[CompileAnalysis] = {
-    analyses.collectFirst {
-      case a if a.relations.productClassName._2s.contains(binaryClassName) => a
-    }
-  }
+  override def lookupAnalysis(binaryClassName: String): Option[CompileAnalysis] =
```

Comment:
why not make this non-optional and provide a default implementation?

Reply:
Empty result here has a meaning - that we have to treat this class as Library dependency instead of External one (so names will be invalidated on any change in *.class or *.jar file).

This change was just optimization since `find` is faster then `collectFirst` (and code looks also better)

Reply:
I thought the empty meant that it falls back to the default impl of blindly using SHA-1. Couldn't that default impl just be moved to be the default?